sql/log_event_server.cc

```diff
@@ -5803,6 +5803,25 @@ int Rows_log_event::do_add_row_data(uchar *row_data, size_t length)
 
 #if defined(HAVE_REPLICATION)
 
+void Rows_log_event::make_thd_info(thd_info_cache cache,
+  const char *class_name, std::initializer_list<const char *> method_names)
+{
+  const LEX_CSTRING &table_name= m_table->s->table_name;
+  int table_name_length= static_cast<int>(table_name.length);
+  char quote_char=
+    get_quote_char_for_identifier(thd, table_name.str, table_name.length);
+  for (const char *method_name: method_names)
+    my_snprintf(*(cache++), sizeof(*cache), "%s::%s("
+      #ifdef WSREP_PROC_INFO
+        "%lld"
+      #endif
+    ") on table %c%.*s%c", class_name, method_name,
+      #ifdef WSREP_PROC_INFO
+        wsrep_thd_trx_seqno(thd),
+      #endif
+    quote_char, table_name_length, table_name.str, quote_char);
```

Comment:
I found that all `thd_proc_info()`s are in `Rows_log_event::do_exec_row()` and have the same format, including the WSRep variants.